From 77966a0d914d1ff94b155d239b49917ff8a50f08 Mon Sep 17 00:00:00 2001
From: Jakub Jelinek <jakub@redhat.com>
Date: Mon, 17 Jul 2017 21:41:08 +0200
Subject: [PATCH] =?utf8?q?backport:=20re=20PR=20sanitizer/81066=20(sanitiz?=
 =?utf8?q?er=5Fstoptheworld=5Flinux=5Flibcdep.cc:276:22:=20error:=20aggreg?=
 =?utf8?q?ate=20=E2=80=98sigaltstack=20handler=5Fstack=E2=80=99=20has=20in?=
 =?utf8?q?complete=20type=20and=20cannot=20be=20defined)?=
MIME-Version: 1.0
Content-Type: text/plain; charset=utf8
Content-Transfer-Encoding: 8bit

	Backported from mainline
	2017-07-14  Jakub Jelinek  <jakub@redhat.com>

	PR sanitizer/81066
	* sanitizer_common/sanitizer_linux.h: Cherry-pick upstream r307969.
	* sanitizer_common/sanitizer_linux.cc: Likewise.
	* sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc: Likewise.
	* tsan/tsan_platform_linux.cc: Likewise.

From-SVN: r250287
---
 libsanitizer/ChangeLog                                | 11 +++++++++++
 libsanitizer/sanitizer_common/sanitizer_linux.cc      |  3 +--
 libsanitizer/sanitizer_common/sanitizer_linux.h       |  4 +---
 .../sanitizer_stoptheworld_linux_libcdep.cc           |  2 +-
 libsanitizer/tsan/tsan_platform_linux.cc              |  2 +-
 5 files changed, 15 insertions(+), 7 deletions(-)

# diff --git a/libsanitizer/ChangeLog b/libsanitizer/ChangeLog
# index 7cb4257fb91..a407df94f6b 100644
# --- a/libsanitizer/ChangeLog
# +++ b/libsanitizer/ChangeLog
# @@ -1,3 +1,14 @@
# +2017-07-17  Jakub Jelinek  <jakub@redhat.com>
# +
# +	Backported from mainline
# +	2017-07-14  Jakub Jelinek  <jakub@redhat.com>
# +
# +	PR sanitizer/81066
# +	* sanitizer_common/sanitizer_linux.h: Cherry-pick upstream r307969.
# +	* sanitizer_common/sanitizer_linux.cc: Likewise.
# +	* sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc: Likewise.
# +	* tsan/tsan_platform_linux.cc: Likewise.
# +
#  2017-05-02  Release Manager
#  
#  	* GCC 7.1.0 released.
diff --git a/libsanitizer/sanitizer_common/sanitizer_linux.cc b/libsanitizer/sanitizer_common/sanitizer_linux.cc
index 806fcd5e284..5b6f18602e7 100644
--- a/libsanitizer/sanitizer_common/sanitizer_linux.cc
+++ b/libsanitizer/sanitizer_common/sanitizer_linux.cc
@@ -605,8 +605,7 @@ uptr internal_prctl(int option, uptr arg2, uptr arg3, uptr arg4, uptr arg5) {
 }
 #endif
 
-uptr internal_sigaltstack(const struct sigaltstack *ss,
-                         struct sigaltstack *oss) {
+uptr internal_sigaltstack(const void *ss, void *oss) {
   return internal_syscall(SYSCALL(sigaltstack), (uptr)ss, (uptr)oss);
 }
 
diff --git a/libsanitizer/sanitizer_common/sanitizer_linux.h b/libsanitizer/sanitizer_common/sanitizer_linux.h
index 895bfc18195..a42df576405 100644
--- a/libsanitizer/sanitizer_common/sanitizer_linux.h
+++ b/libsanitizer/sanitizer_common/sanitizer_linux.h
@@ -19,7 +19,6 @@
 #include "sanitizer_platform_limits_posix.h"
 
 struct link_map;  // Opaque type returned by dlopen().
-struct sigaltstack;
 
 namespace __sanitizer {
 // Dirent structure for getdents(). Note that this structure is different from
@@ -28,8 +27,7 @@ struct linux_dirent;
 
 // Syscall wrappers.
 uptr internal_getdents(fd_t fd, struct linux_dirent *dirp, unsigned int count);
-uptr internal_sigaltstack(const struct sigaltstack* ss,
-                          struct sigaltstack* oss);
+uptr internal_sigaltstack(const void* ss, void* oss);
 uptr internal_sigprocmask(int how, __sanitizer_sigset_t *set,
     __sanitizer_sigset_t *oldset);
 
diff --git a/libsanitizer/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc b/libsanitizer/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc
index 891386dc0ba..234e8c652c6 100644
--- a/libsanitizer/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc
+++ b/libsanitizer/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc
@@ -273,7 +273,7 @@ static int TracerThread(void* argument) {
 
   // Alternate stack for signal handling.
   InternalScopedBuffer<char> handler_stack_memory(kHandlerStackSize);
-  struct sigaltstack handler_stack;
+  stack_t handler_stack;
   internal_memset(&handler_stack, 0, sizeof(handler_stack));
   handler_stack.ss_sp = handler_stack_memory.data();
   handler_stack.ss_size = kHandlerStackSize;
diff --git a/libsanitizer/tsan/tsan_platform_linux.cc b/libsanitizer/tsan/tsan_platform_linux.cc
index 2ed5718a12e..6f972ab0dd6 100644
--- a/libsanitizer/tsan/tsan_platform_linux.cc
+++ b/libsanitizer/tsan/tsan_platform_linux.cc
@@ -287,7 +287,7 @@ void InitializePlatform() {
 int ExtractResolvFDs(void *state, int *fds, int nfd) {
 #if SANITIZER_LINUX && !SANITIZER_ANDROID
   int cnt = 0;
-  __res_state *statp = (__res_state*)state;
+  struct __res_state *statp = (struct __res_state*)state;
   for (int i = 0; i < MAXNS && cnt < nfd; i++) {
     if (statp->_u._ext.nsaddrs[i] && statp->_u._ext.nssocks[i] != -1)
       fds[cnt++] = statp->_u._ext.nssocks[i];
-- 
2.18.4
